catalog/app/components/FileEditor/routes.ts

```diff
@@ -15,38 +14,52 @@ interface RouteMap {
   bucketPackageDetail: Routes.BucketPackageDetailArgs
 }
 
-export function editFileInPackage(
-  urls: NamedRoutes.Urls<RouteMap>,
-  handle: Model.S3.S3ObjectLocation,
-  logicalKey: string,
-  next: string,
-) {
-  return urls.bucketFile(handle.bucket, handle.key, {
-    add: logicalKey,
-    edit: true,
-    next,
-  })
-}
-
 export function useEditFileInPackage(
   packageHandle: PackageHandle,
   fileHandle: Model.S3.S3ObjectLocation,
-  logicalKey: string,
-) {
+): (logicalKey: string) => string {
+  const { urls } = NamedRoutes.use<RouteMap>()
+  return React.useCallback(
+    (logicalKey: string) =>
+      urls.bucketFile(fileHandle.bucket, fileHandle.key, {
+        add: logicalKey,
+        edit: true,
+        next: urls.bucketPackageDetail(packageHandle.bucket, packageHandle.name, {
+          action: 'revisePackage',
+        }),
+      }),
+    [fileHandle, packageHandle, urls],
```

Comment:
i'd probably destructure the handle objects and use individual props explicitly for memoization to make it somewhat more foolproof (like you do in `useAddFileInPackage`)

Reply:
I subscribed to `packageHandle` in `useAddFileInPackage` as well.